src/cryptsetup/cryptsetup-generator.c

```diff
@@ -393,22 +393,37 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                r = sscanf(value, "%m[0-9a-fA-F-]=%ms", &uuid, &uuid_value);
-                if (r == 2) {
+                r = sscanf(value, "%m[0-9a-fA-F-]", &uuid);
```

Comment:
i'd feel happier if we'd not use sscanf() for this, but use extract_first_word() instead...

Reply:
Hmm, ok, but ```scanf()``` is used multiple times. I'd rather post follow-up PR where I will refactor the code to use ```extract_first_word()```.

Reply:
sure